### DISCUSSION TITLE
Is it possible to use MSBuild properties to populate the project matrix of a file?

### DISCUSSION THREAD

**Discussion by Arthri** [dotnet/project-system#9534] [Q&A]
Visual Studio has a combobox for every file, which can be used to switch between the different "contexts"(projects) that include that file

![ComboBox](https://github.com/user-attachments/assets/1a82298d-c836-48fd-98c4-6a2349ac297f)

Captured in the above image is the combobox along with some sample values it can have. Currently, I know that new contexts can be as follows,
- Add a new target framework to a multi-TFM project (`ProjectA`)
- Include the file in another project through `<Compile Include"..." />` (`ProjectB`)

I wish to know if there's other ways to create such contexts, for example, having `ProjectA (x86)` and `ProjectA (x64)` compiled and checked side-by-side(like `ProjectA (net7.0)` and `ProjectA (net8.0)`) without having to switch the Platform using the build configuration manager. I hope to know if there's way to extend it with any arbitrary MSBuild property, not just `Platform`

**Comment by drewnoakes** [comment]
Does Visual Studio's "Batch Build" feature do what you want?

![image](https://github.com/user-attachments/assets/2cb272a5-5360-428e-9241-6d94feff4e76)

You would check the set of configurations you wish to build.

This UI is available via the _Build_ menu.

**Reply by Arthri** [reply]
Thanks for the suggestion! Although, I don't think that's what I'm looking for, because IntelliSense doesn't load them in parallel(as in the combobox) and they seem to be limited to the Platform and Configuration properties, whereas I'm looking for a solution which can use arbitrary MSBuild properties
